### ISSUE TITLE
when moving horizontally in the code editor using trackpad on macOS, the code moves to the opposite direction.

### ISSUE THREAD

**Issue by Wang-Yue** [nesbox/TIC-80#2385]
when moving horizontally in the code editor using trackpad on macOS, the code moves to the opposite direction.

This is a similar to https://github.com/nesbox/TIC-80/issues/2335

**Comment by TimotheeGreg** [comment]
Have you tested with the [nightly build](https://nightly.link/nesbox/TIC-80/workflows/build/main) ? This should have been fixed with #2358 ?
Note: #2355 was an erroneous fix of #2335 which was partly reversed and fixed with #2358.

**Comment by Wang-Yue** [comment]
As far as I understand, 2358 (and 2376) reversed 2335 (2355) and added the handling when user is moving the mouse while pressing the shift key. (see the changes in src/studio/editors/code.c under the `if(tic_api_key(tic, tic_key_shift))` test). 

The bug I reported is not related to the scroll direction when "moving mouse wheel and holding shift", but when "moving horizontally using a trackpad". When you swipe left using the trackpad, the code should move to the left (which is the default native experience of most apps on macOS, ranging from TextEdit, Safari to Google Chrome), not right (current tic-80 behavior).

**Comment by TimotheeGreg** [comment]
Oh, indeed! My mistake, sorry!
In fact, I realize that #2358 (and #2376) did not solve the issue. I probably tested too quickly.
I was asking to correct this with the trackpad too [here](https://github.com/nesbox/TIC-80/issues/2335#issuecomment-1776679711). And "moving mouse wheel and holding shift" was the only case for which there was no need of a correction as written [here](https://github.com/nesbox/TIC-80/issues/2335#issue-1957599087).

So to summarize, today the horizontal scrolling is incorrect for:
* trackpads
* side movements of mouse wheels that allow such movements

**Comment by Wang-Yue** [comment]
I just tested the following patch

```
diff --git a/src/studio/studio.c b/src/studio/studio.c
index 4c202a19..29433ce9 100644
--- a/src/studio/studio.c
+++ b/src/studio/studio.c
@@ -2141,6 +2141,7 @@ static void processMouseStates(Studio* studio)
 
         state->dbl.ticks++;
     }
+    tic->ram->input.mouse.scrollx *= -1;
 }
 
 static void blitCursor(Studio* studio)
```

should solve this issue. just don't know if it will break other things. 
adding an option like #2335 / #2355 sounds safer. that PR's problem was it mistakes scrollx with scrolly.
